tools/impl/release_parsing.py

```diff
@@ -66,32 +66,43 @@ def _parse_build_names(file_path: Path) -> ReleaseInfo:
         # Strip and get non-empty lines
         parsed_file_names = [line.strip() for line in handle if line.strip()]
 
-    # All non-nightly files have the date as their first line
+    # All non-nightly files have the date as their first line and they will all be on the same date
     if not is_nightly:
         release_date = datetime.date.fromisoformat(parsed_file_names.pop(0))
+        # For an official release we have a single ReleaseInfo and multiple packages
+        package_details: List[PackageDetails] = []
+        for executable_name in parsed_file_names:
+            package_details.append(
+                PackageDetails(os_details=get_os(executable_name, version),
+                               download_url=get_download_url(executable_name, version, is_nightly)))
+        releases = [
+            ReleaseInfo(date=release_date,
+                        version=version,
+                        formatted_version=formatted_version,
+                        package_details=package_details)
+        ]
     else:
-        release_date = get_nightly_date(parsed_file_names[0])
-
-    package_details: List[PackageDetails] = []
-
-    for executable_name in parsed_file_names:
-        package_details.append(
-            PackageDetails(os_details=get_os(executable_name, version),
-                           download_url=get_download_url(executable_name, version, is_nightly))
-        )
-
-    return ReleaseInfo(date=release_date, version=version,
-                       formatted_version=formatted_version, package_details=package_details)
-
-
-def get_nightly_release() -> ReleaseInfo:
+        # For nightlies we have separate releases per file as sometimes not all builds complete
+        releases: List[PackageDetails] = []
```

Comment:
This type is incorrect, since we append `ReleaseInfo` types to this list instead of `PackageDetails`

Reply:
Oh yes. Mistyped. Thank you!